lib/logstash/util/accessors.rb

```diff
@@ -48,7 +48,9 @@ def strict_set(accessor, value)
 
     def del(accessor)
       target, key = lookup(accessor)
-      target.is_a?(Array) ? target.delete_at(key.to_i) : target.delete(key)
+      unless target.nil?
```

Comment:
good point @jordansissel about the nil vs false, as this is what happen we should always check for nil :-)